### ISSUE TITLE
Is extDB2 supported on Windows 10?

### ISSUE THREAD

**Issue by moosemilker** [Torndeco/extDB2#90]
In Windows 10, I cannot get the extDB2.dll to work with ExileMod, after following the instructions 100%.  I had this working with Windows 8.1.

I installed the x86 VS 2013 Redistributable, which has no change on extDB.

I get this in the RPT:

11:47:13 "ExileServer - Server is loading..."
11:47:14 "ExileServer - MySQL connection error!"
11:47:14 "ExileServer - MySQL Error: Unable to locate extDB2 extension!"
11:47:14 "ExileServer - Server will shutdown now :("

**Comment by chris579** [comment]
MySQL connection error means the server can't connect to your mySQL database. You should check if the connection properties entered in your extdb-conf.ini are correct.

**Comment by Torndeco** [comment]
You didn't read the readme for ExileServer
Move the tbb+tbbmalloc to Arma 3 Root Directory

**Comment by moosemilker** [comment]
Those files are in the root directory...

**Comment by chris579** [comment]
Where is your extDB2.dll located?

**Comment by moosemilker** [comment]
It's in .\@ExileServer\, just as the ExileMod server download is packaged.

I got further.  My start script had "-malloc=tbbmalloc" which was bypassing the Arma 3 root tbb*.dll files... So Torndeco was partially correct, since that switch was causing the Arma server to look in the Arma root .\dll\ directory for the tbb*.dll files.

Now I am further along to the requesting Bambi Creation step on the server (happens after you pick a spawn location).  I checked the database, and it still is not writing rows to the database yet.  I'm looking further into this now.

RPT excerpt (with my ID purposefully replaced with 'XXXREMOVED_IDXXX'):


17:02:42 "ExileServer - Server is loading..."
17:02:42 "ExileServer - Installed extDB2 version: 65"
17:02:42 "ExileServer - Connected to database!"
17:02:42 "ExileServer - Database protocol initialized!"
17:02:42 "Terd Fergasen/BIS_fnc_log: [preInit] ExileServer_fnc_preInit (78.9948 ms)"
17:02:43 Error: Bone world root doesn't exist in skeleton OFP2_ManSkeleton
17:02:43 Error: Bone hips doesn't exist in skeleton OFP2_ManSkeleton
17:02:47 "Terd Fergasen/BIS_fnc_log: [script] initServer.sqf"
17:02:47 "Terd Fergasen/BIS_fnc_log: [postInit] BIS_fnc_missionFlow (0 ms)"
17:02:47 "Terd Fergasen/BIS_fnc_log: [postInit] BIS_fnc_initParams (0 ms)"
17:02:47 "Terd Fergasen/BIS_fnc_log: [postInit] BIS_fnc_initRespawn (0 ms)"
17:02:47 "Terd Fergasen/BIS_fnc_log: [postInit] BIS_fnc_sharedObjectives (0 ms)"
17:02:47 "Terd Fergasen/BIS_fnc_log: [postInit] BIS_fnc_3dMarkers (0 ms)"
17:02:47 "Terd Fergasen/BIS_fnc_log: [postInit] ExileClient_fnc_postInit (0 ms)"
17:02:47 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:47 "ExileServer - Job with handle 0 added."
17:02:47 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:47 "ExileServer - Job with handle 1 added."
17:02:47 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:47 "ExileServer - Job with handle 2 added."
17:02:47 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:47 "ExileServer - Job with handle 3 added."
17:02:47 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:47 "ExileServer - Job with handle 4 added."
17:02:47 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:47 "ExileServer - Job with handle 5 added."
17:02:47 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:48 c:\w\stable\futura\lib\network\networkserver.cpp NetworkServer::OnClientStateChanged:NOT IMPLEMENTED - briefing!
17:02:48 "ExileServer - Job with handle 6 added."
17:02:48 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:48 "ExileServer - Job with handle 7 added."
17:02:48 Weather was forced to change
17:02:48 "ExileServer - Main thread Delay adjusted to 1 seconds "
17:02:48 "ExileServer - Job with handle 8 added."
17:02:48 "ExileServer - Initializing game world..."
17:02:48 "ExileServer - Loading territories from database..."
17:02:48 "ExileServer - Done loading territories!"
17:02:48 "ExileServer - Loading constructions from database..."
17:02:48 "ExileServer - Done loading constructions!"
17:02:48 "ExileServer - Loading vehicles from database..."
17:02:48 "ExileServer - Done loading vehicles!"
17:02:48 "ExileServer - Loading containers from database..."
17:02:48 "ExileServer - Done loading containers!"
17:02:48 "ExileServer - Creating spawn zone vehicles..."
17:02:50 "ExileServer - Spawning Dynamic Vehicles. GridSize: 2200 Vehs/Grid : 2"
17:02:50 "ExileServer - onPlayerConnected : ["XXXREMOVED_IDXXX","Terd Fergasen"]"
17:02:50 "ExileServer - Player Terd Fergasen (UID XXXREMOVED_IDXXX) connected!"
17:02:50 "ExileServer - onPlayerConnected : ["","__SERVER__"]"
17:02:51 Error: Object(3 : 5) not found
17:02:51 Exile_Cosmetic_MG: MainTurret - unknown animation source mainTurret
17:02:51 Exile_Cosmetic_MG: MainGun - unknown animation source mainGun
17:02:51 Exile_Cosmetic_MG: Ammo_belt_rotation - unknown animation source ReloadAnim
17:02:51 Exile_Cosmetic_MG: Bolt_reload_begin - unknown animation source reloadMagazine
17:02:51 Exile_Cosmetic_MG: muzzleFlash - unknown animation source muzzle_source
17:02:51 Exile_Cosmetic_MG: zaslehROT - unknown animation source muzzle_source_rot
17:02:51 Exile_Cosmetic_MG: AddAutonomous_unhide - unknown animation source autonomous_unhide
17:02:51 Exile_Cosmetic_MG: bullet001_reload_hide - unknown animation source revolving
17:02:51 Error: Object(3 : 21) not found
17:02:51 Error: Object(3 : 22) not found
17:02:51 Error: Object(3 : 23) not found
17:02:51 Error: Object(3 : 24) not found
17:02:51 Error: Object(3 : 25) not found
17:02:51 Error: Object(3 : 26) not found
17:02:51 Exile_Cosmetic_UAV: RotorImpactHide - unknown animation source rotorHFullyDestroyed
17:02:51 Exile_Cosmetic_UAV: TailRotorImpactHide - unknown animation source tailRotorHFullyDestroyed
17:02:51 Exile_Cosmetic_UAV: propeller1_rotation - unknown animation source rotorH
17:02:51 Exile_Cosmetic_UAV: propeller2_rotation - unknown animation source rotorV
17:02:51 Exile_Cosmetic_UAV: propeller1_hide - unknown animation source rpm
17:02:51 Exile_Cosmetic_UAV: MainTurret - unknown animation source mainTurret
17:02:51 Exile_Cosmetic_UAV: MainGun - unknown animation source mainGun
17:03:03 "ExileServer - Dynamic vehicles spawned. Count : 234"
17:03:03 "ExileServer - Game world initialized! Let the fun begin!"
17:03:03 "ExileServer - Server is up and running! Version: 0.9.18"
17:03:03 "Terd Fergasen/BIS_fnc_log: [postInit] ExileServer_fnc_postInit (16021 ms)"
17:03:03 "ExileServer - Main thread started"
17:03:04 "ExileServer - Session ID for 'Terd Fergasen' is 'XEYrUbXM'!"
17:03:04 "Terd Fergasen/BIS_fnc_log: [BIS_fnc_preload] ----- Scripts initialized at 21444 ms -----"
17:03:09 "["ItemCompass",4]"
17:03:09 "["ItemMap",4]"
17:03:09 "["Exile_Item_XM8",4]"
17:03:09 "["ItemRadio",4]"
17:03:09 "["Exile_Item_PlasticBottleFreshWater",1]"
17:03:13 Error in expression <me = name _requestingPlayer;
_clanID = (_accountData select 4);
_clanName = (_ac>
17:03:13   Error position: <_accountData select 4);
_clanName = (_ac>
17:03:13   Error Undefined variable in expression: _accountdata
17:03:13 File exile_server\code\ExileServer_object_player_createBambi.sqf, line 22
17:03:13 Error in expression <rmat["%1", _result];
(_result select 1) select 0>
17:03:13   Error position: <select 0>
17:03:13   Error Generic error in expression
17:03:13 File exile_server\code\ExileServer_system_database_query_insertSingle.sqf, line 15
17:03:49 "ExileServer - OnHandleDisconnect: [R Alpha 1-2:1,4.12695e+008,"XXXREMOVED_IDXXX","Terd Fergasen"]"
17:03:49 Warning: Cleanup player - person 2:1 not found

**Comment by Torndeco** [comment]
Go to exile forums for support.
Am afraid this github is just for extDB2, i cant really be providing specfic support for multiple missions / mods. It will just become a mess

**Comment by moosemilker** [comment]
Thanks. Yep. I am already looking through their forums.  Just closed the loop for people asking questions.  extDB looks to be working fine, since this is in the RPT:

17:02:42 "ExileServer - Server is loading..."
17:02:42 "ExileServer - Installed extDB2 version: 65"
17:02:42 "ExileServer - Connected to database!"
17:02:42 "ExileServer - Database protocol initialized!"

**Comment by moosemilker** [comment]
For anyone that reads this thread, hopefully this helps you...

I fixed my own issue…

From extDB log: [mysql_stmt_error]: Operation not allowed when innodb_forced_recovery > 0

I forgot I had a database with an issue and had ‘innodb_forced_recovery=1’… I dropped the database in a bad state, removed the line from the my.ini file, restarted the MySQL service and the game loads/plays just fine.

Thanks to the developers of extDB and ExileMod! Good stuff!

-Aaron
